### 1. Is there a correlation between postal votes % and voter turnout %?

In [1]:
import matplotlib.pyplot as plt
import pandas as pd

# read data from excel file
data = pd.read_excel('C:\\Users\HP\Desktop\\2.xlsx')

# display the contents
data.head()

pc_name  voter_turnout_ratio  postal_vote_percent
0    Adilabad                 75.44                62.63
1  Peddapalle                 71.71                 4.71
2  Karimnagar                 72.55                17.90
3    Nizamabad                69.10                33.12
4    Zahirabad                75.75                58.99

In [2]:
# compute the correlation matrix
corr_matrix = data.corr()
corr_matrix

voter_turnout_ratio  postal_vote_percent
voter_turnout_ratio             1.000000            -0.032101
postal_vote_percent            -0.032101             1.000000

The correlation coefficient between voter_turnout_ratio and postal_vote_percent is -0.032101. This value is very close to 0, indicating that there is almost no linear relationship between these two variables. Therefore, based on this correlation coefficient, one can conclude that there is no significant correlation between voter_turnout_ratio and postal_vote_percent.

### 2019 

In [3]:
# read the excel file 
data_19= pd.read_excel('C:\\Users\HP\Desktop\\scatter_plot_2019.xlsx')

data_19.head()

pc_name  voter_turnout_ratio  postal_vote_percent
0         Aruku                 74.03                25.63
1     Srikakulam                74.48                37.23
2   Vizianagaram                80.98                34.24
3  Visakhapatnam                67.77                 6.87
4     Anakapalli                81.41                29.47

In [4]:
# compute the correlation matrix
corr_matrix_19 = data_19.corr()
corr_matrix_19

voter_turnout_ratio  postal_vote_percent
voter_turnout_ratio              1.00000             -0.10209
postal_vote_percent             -0.10209              1.00000

There is a very weak negative correlation between voter_turnout_ratio and postal_vote_percent, suggesting minimal linear relationship between the two variables (the coefficient is still quite low and close to 0).

### 2. Is there any correlation between GDP of a state and voter turnout %?

In [5]:
gdp = pd.read_excel('C:\\Users\HP\Desktop\\gdp.xlsx')
gdp.head()

state  2013_14  2018_19
0     Andhra Pradesh   4642720  8737210
1  Arunachal Pradesh    145810   253350
2              Assam   1777450  3093360
3              Bihar   3171010  5279760
4       Chhattisgarh   2068330  3271070

In [6]:
# as using the data for 2014 analysis, drop the 2018-19 column. If using for 2019 analysis, drop the column for 2013-14 for the sake of simplicity
df = gdp.drop('2018_19', axis=1)
df.head()

state  2013_14
0     Andhra Pradesh   4642720
1  Arunachal Pradesh    145810
2              Assam   1777450
3              Bihar   3171010
4       Chhattisgarh   2068330

In [7]:
# read the excel file
v_14 = pd.read_excel('C:\\Users\HP\Desktop\\state_vtr_14.xlsx')
v_14.head()

state  state_voter_turnout_ratio
0  Andaman & Nicobar Islands                      70.66
1             Andhra Pradesh                      77.67
2          Arunachal Pradesh                      78.61
3                      Assam                      80.08
4                      Bihar                      56.25

In [8]:
# standardize state names by stripping extra spaces and consistent formatting
df['state'] = df['state'].str.strip().str.replace('&', 'and').str.replace('\s*\(.*\)', '', regex=True)
v_14['state'] = v_14['state'].str.strip().str.replace('&', 'and').str.replace('\s*\(.*\)', '', regex=True)


In [9]:
# merge the dataframes using the common column in both the dataframes (performing inner join)
merged_df = pd.merge(df,v_14,on = "state")
merged_df.head()

state  2013_14  state_voter_turnout_ratio
0     Andhra Pradesh  4642720                      77.67
1  Arunachal Pradesh   145810                      78.61
2              Assam  1777450                      80.08
3              Bihar  3171010                      56.25
4                Goa   359210                      77.02

In [10]:
# compute the correlation matrix
corr_matr_14 = merged_df.corr()
corr_matr_14

2013_14  state_voter_turnout_ratio
2013_14                    1.000000                  -0.329826
state_voter_turnout_ratio -0.329826                   1.000000

The correlation coefficient of -0.329826 suggests a moderate negative correlation between gdp of 2013_14 and state_voter_turnout_ratio. This means that as the value of one variable increases, the value of the other variable tends to decrease, but the relationship is not very strong.

### 2019

In [11]:
# for 2019 analysis, drop the 2013-14 column for simplicity
df_19 = gdp.drop('2013_14', axis = 1)
df_19.head()

state  2018_19
0     Andhra Pradesh   8737210
1  Arunachal Pradesh    253350
2              Assam   3093360
3              Bihar   5279760
4       Chhattisgarh   3271070

In [12]:
# read the excel file with state voter turnout % for 2019
v_19 = pd.read_excel('C:\\Users\HP\Desktop\\state_vtr_19.xlsx')
v_19.head()

state  state_voter_turnout_ratio
0     Andhra Pradesh                      80.22
1  Arunachal Pradesh                      81.15
2              Assam                      81.57
3              Bihar                      57.30
4                Goa                      75.10

In [13]:
# standardize state names by stripping extra spaces and consistent formatting before merging the dataframes
df_19['state'] = df_19['state'].str.strip().str.replace('&', 'and').str.replace('\s*\(.*\)', '', regex=True)
v_19['state'] = v_19['state'].str.strip().str.replace('&', 'and').str.replace('\s*\(.*\)', '', regex=True)


In [14]:
# merge the dataframes
merged_df_19 = pd.merge(df_19,v_19,on = "state")
merged_df_19.head()

state  2018_19  state_voter_turnout_ratio
0     Andhra Pradesh  8737210                      80.22
1  Arunachal Pradesh   253350                      81.15
2              Assam  3093360                      81.57
3              Bihar  5279760                      57.30
4       Chhattisgarh  3271070                      71.59

In [15]:
# compute the correlation matrix
correlation_matrix_19 = merged_df_19.corr()
correlation_matrix_19

2018_19  state_voter_turnout_ratio
2018_19                    1.00000                   -0.27919
state_voter_turnout_ratio -0.27919                    1.00000

The correlation coefficient of -0.27919 indicates a weak negative correlation. This suggests that as the value of one variable increases, the value of the other variable tends to decrease, but the relationship is relatively weak.

### 3. Is there any correlation between literacy % of a state and voter turnout %?

In [16]:
# read the excel file containing literacy rate data
lr = pd.read_excel('C:\\Users\HP\Desktop\\literacy_rate.xlsx')
lr.head()

state  Census_2011  NSO_survey_2017
0  Andaman & Nicobar Islands        86.63            86.27
1             Andhra Pradesh        67.02            66.90
2          Arunachal Pradesh        65.38            66.95
3                      Assam        72.19            85.90
4                      Bihar        61.80            70.90

In [17]:
# for 2014 analysis, dropping the other column
lr_14 = lr.drop('NSO_survey_2017',axis = 1)

In [19]:
lr_merge_14 = pd.merge(lr_14,v_14,on = 'state')
lr_merge_14.head()

state  Census_2011  state_voter_turnout_ratio
0     Andhra Pradesh        67.02                      77.67
1  Arunachal Pradesh        65.38                      78.61
2              Assam        72.19                      80.08
3              Bihar        61.80                      56.25
4         Chandigarh        86.05                      73.71

In [20]:
# compute the correlation matrix
c_matrix_14 = lr_merge_14.corr()
c_matrix_14

Census_2011  state_voter_turnout_ratio
Census_2011                   1.000000                   0.240189
state_voter_turnout_ratio     0.240189                   1.000000

The correlation coefficient of 0.240189 indicates a weak positive correlation. This suggests that as the value of one variable increases, the value of the other variable also tends to increase, but the relationship is relatively weak.

### 2019

In [21]:
# For 2019 analysis, dropping the mentioned column
lr_19 = lr.drop('Census_2011',axis = 1)
lr_19.head()

state  NSO_survey_2017
0  Andaman & Nicobar Islands            86.27
1             Andhra Pradesh            66.90
2          Arunachal Pradesh            66.95
3                      Assam            85.90
4                      Bihar            70.90

In [22]:
# merge the dataframes
merge_19 = pd.merge(lr_19,v_19, on = 'state')

In [23]:
# compute the correlation matrix
c_matrix_19 = merge_19.corr()
c_matrix_19

NSO_survey_2017  state_voter_turnout_ratio
NSO_survey_2017                   1.000000                   0.204556
state_voter_turnout_ratio         0.204556                   1.000000

The correlation coefficient of 0.204556 indicates a weak positive correlation. This suggests that as the value of one variable increases, the value of the other variable also tends to increase, but the relationship is relatively weak.